generators/typescript/sdk/client-class-generator/src/GeneratedWebsocketClientClassImpl.ts

```diff
@@ -0,0 +1,396 @@
+        };
+
+        const serviceClass: ClassDeclarationStructure = {
+            kind: StructureKind.Class,
+            name: this.serviceClassName,
+            isExported: true,
+            properties: [],
+            ctors: [],
+            methods: []
+        };
+
+        serviceClass.ctors?.push({
+            parameters: [
+                {
+                    name: GeneratedWebsocketClientClassImpl.OPTIONS_PRIVATE_MEMBER,
+                    isReadonly: true,
+                    scope: Scope.Protected,
+                    type: `${this.serviceClassName}.${GeneratedWebsocketClientClassImpl.OPTIONS_INTERFACE_NAME}`
+                }
+            ]
+        });
+
+        const statements = this.generateConnectMethodStatements(context);
+
+        serviceClass.methods?.push({
+            name: "connect",
+            scope: Scope.Public,
+            isAsync: true,
+            parameters: [
+                {
+                    name: "args",
+                    type: `${this.serviceClassName}.${GeneratedWebsocketClientClassImpl.CONNECT_ARGS_INTERFACE_NAME}`,
+                    initializer: "{}"
+                }
+            ],
+            returnType: getTextOfTsNode(
+                ts.factory.createTypeReferenceNode(ts.factory.createIdentifier("Promise"), [
+                    this.getSocketTypeNode(context)
+                ])
+            ),
+            statements: statements.map(getTextOfTsNode)
+        });
+
+        context.sourceFile.addModule(serviceModule);
+        context.sourceFile.addClass(serviceClass);
+    }
+
+    private generateOptionsInterface(context: SdkContext): InterfaceDeclarationStructure {
+        const generatedEnvironments = context.environments.getGeneratedEnvironments();
+        return {
+            kind: StructureKind.Interface,
+            name: GeneratedWebsocketClientClassImpl.OPTIONS_INTERFACE_NAME,
+            properties: [
+                {
+                    name: GeneratedWebsocketClientClassImpl.ENVIRONMENT_OPTION_PROPERTY_NAME,
+                    type: getTextOfTsNode(
+                        context.coreUtilities.fetcher.Supplier._getReferenceToType(
+                            generatedEnvironments.getTypeForUserSuppliedEnvironment(context)
+                        )
+                    ),
+                    hasQuestionToken: generatedEnvironments.hasDefaultEnvironment(),
+                    docs: ["The environment to use for the websocket. Defaults to the default environment."]
+                },
+                {
+                    name: GeneratedWebsocketClientClassImpl.BASE_URL_OPTION_PROPERTY_NAME,
+                    type: getTextOfTsNode(
+                        context.coreUtilities.fetcher.Supplier._getReferenceToType(
+                            ts.factory.createKeywordTypeNode(ts.SyntaxKind.StringKeyword)
+                        )
+                    ),
+                    hasQuestionToken: this.channel.baseUrl != null,
```

Comment:
Should this _always_ be optional if it's meant to act as an override (re: the comment below)?